felix/fv/wireguard_3node_test.go

```diff
@@ -0,0 +1,289 @@
+// Copyright (c) 2020-2022 Tigera, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build fvtests
+
+package fv_test
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"regexp"
+	"strconv"
+	"strings"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/types"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/projectcalico/calico/felix/fv/connectivity"
+	"github.com/projectcalico/calico/felix/fv/containers"
+	"github.com/projectcalico/calico/felix/fv/infrastructure"
+	"github.com/projectcalico/calico/felix/fv/tcpdump"
+	"github.com/projectcalico/calico/felix/fv/utils"
+	"github.com/projectcalico/calico/felix/fv/workload"
+	"github.com/projectcalico/calico/libcalico-go/lib/apiconfig"
+	"github.com/projectcalico/calico/libcalico-go/lib/clientv3"
+)
+
+var (
+	dataStoreTypes = []apiconfig.DatastoreType{apiconfig.EtcdV3, apiconfig.Kubernetes}
+)
+
+type wireguard3NodeTestConf struct {
+	routeSource           string
+	ipipEnabled           bool
+	borrowedIPs           bool
+	hostEncryptionEnabled bool
+}
```

Comment:
Think this needs the good ol `wireguardEnabled` flag too

Reply:
for these tests, WireGuard is required to be enabled all the time and it's already hard-coded in wireguardTopologyOptions